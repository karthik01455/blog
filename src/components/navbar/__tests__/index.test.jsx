import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import Navbar from '../../navbar';
jest.mock("../../../utils/makeRequests/");
jest.mock("react-router-dom", () => ({
  useNavigate: () => jest.fn(),
}));
describe('Navbar', () => {
  it('should render the same snapshot when it is called', () => {
    const { asFragment } = render(<Navbar />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should navigate to blog page when clicked on contact', () => {
    const navigate = jest.fn();
    const { queryByTestId } = render(<Navbar />);
    const contact = queryByTestId('Contact');
    fireEvent.click(contact);
    // expect(navigate).toHaveBeenCalled(1);
    expect(navigate).toHaveBeenCalledWith('/home/blog');

  });
  it('should navigate to about page when clicked on about', () => {
    const navigate = jest.fn();
    const { queryByText } = render(<Navbar />);
    const about = queryByText('About')
    fireEvent.click(about);
    //expect(navigate).toHaveBeenCalledWith('/home/about');

  });
});
