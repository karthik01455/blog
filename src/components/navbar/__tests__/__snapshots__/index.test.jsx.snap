// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render the same snapshot when it is called 1`] = `
<DocumentFragment>
  <div
    class="nav"
  >
    <ul>
      <li>
        Blog
      </li>
      <li>
        About
      </li>
      <li>
        Contact
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
